File: parse_players.py
import csv
import json
import re

def convert_gdrive_url(url):
    if not url or "drive.google.com" not in url:
        return url
    # Extract file ID from drive.google.com/open?id=... or drive.google.com/file/d/...
    match = re.search(r'[?&]id=([a-zA-Z0-9_-]+)', url)
    if not match:
        match = re.search(r'/file/d/([a-zA-Z0-9_-]+)', url)
    
    if match:
        file_id = match.group(1)
        return f"https://drive.google.com/thumbnail?id={file_id}&sz=w1000"
    return url

players = []
with open('c:/D_drive/RPL-Auction/Form Responses 1.csv', 'r', encoding='utf-8') as f:
    # Custom parsing to handle the multi-line header
    lines = f.readlines()
    # Header is lines[0] and lines[1]. Data starts from lines[2].
    # Let's use csv.reader on the rest of the file.
    data_content = "".join(lines[2:])
    reader = csv.reader(data_content.splitlines())
    
    for row in reader:
        if not row or len(row) < 8:
            continue
        
        name = row[1].strip()
        if not name or name == "Player Name":
            continue
            
        sports_raw = row[2].strip()
        # Handle "Cricket;Volleyball" or "Cricket, Volleyball"
        sport_list = [s.strip() for s in re.split(r'[;,]', sports_raw) if s.strip()]
        
        gender = row[3].strip()
        image_url = convert_gdrive_url(row[4].strip())
        achievements = row[5].strip()
        role = row[6].strip()
        base_price = int(row[7].strip()) if row[7].strip().isdigit() else 100
        
        # Combine role and achievements
        combined_achievements = f"Role: {role} | {achievements}" if role and role != "Nil" else achievements
        
        players.append({
            "name": name,
            "sport": sport_list,
            "gender": gender,
            "basePrice": base_price,
            "image": image_url,
            "achievements": combined_achievements,
            "status": "Waiting"
        })

# Sort by gender: Female first, then Male
# Using alphabetical order ("Female" < "Male")
players.sort(key=lambda p: (p["gender"] != "Female", p["name"]))

print(json.dumps(players, indent=2))

File: parse_players.js
const fs = require('fs');

function convertGDriveUrl(url) {
    if (!url || !url.includes("drive.google.com")) return url;
    const match = url.match(/[?&]id=([a-zA-Z0-9_-]+)/) || url.match(/\/file\/d\/([a-zA-Z0-9_-]+)/);
    if (match) {
        return `https://drive.google.com/thumbnail?id=${match[1]}&sz=w1000`;
    }
    return url;
}

const csvData = fs.readFileSync('c:/D_drive/RPL-Auction/Form Responses 1.csv', 'utf8');
const lines = csvData.split(/\r?\n/);
const dataLines = lines.slice(2); // Skip multi-line header

const players = [];

for (const line of dataLines) {
    if (!line.trim()) continue;
    
    // Simple CSV split (handling some quotes)
    // For a more robust solution we'd use a library, but let's try this simple regex
    const row = line.match(/(".*?"|[^",\s]+)(?=\s*,|\s*$)/g) || [];
    // Rewriting the split to be more reliable for this specific CSV
    // Actually, let's just use a basic split and manually fix the quoted ones
    const parts = [];
    let currentPart = "";
    let inQuotes = false;
    for (let i = 0; i < line.length; i++) {
        const char = line[i];
        if (char === '"') {
            inQuotes = !inQuotes;
        } else if (char === ',' && !inQuotes) {
            parts.push(currentPart.trim());
            currentPart = "";
        } else {
            currentPart += char;
        }
    }
    parts.push(currentPart.trim());

    if (parts.length < 8) continue;

    const name = parts[1].replace(/^"|"$/g, '');
    if (!name || name === "Player Name") continue;

    const sportsRaw = parts[2].replace(/^"|"$/g, '');
    const sportList = sportsRaw.split(/[;,]/).map(s => s.trim()).filter(Boolean);

    const gender = parts[3].replace(/^"|"$/g, '');
    const imageUrl = convertGDriveUrl(parts[4].replace(/^"|"$/g, ''));
    const achievements = parts[5].replace(/^"|"$/g, '');
    const role = parts[6].replace(/^"|"$/g, '');
    const basePrice = parseInt(parts[7].replace(/^"|"$/g, '')) || 100;

    const combinedAchievements = (role && role !== "Nil") ? `Role: ${role} | ${achievements}` : achievements;

    players.push({
        name: name,
        sport: sportList,
        gender: gender,
        basePrice: basePrice,
        image: imageUrl || null,
        achievements: combinedAchievements,
        status: "Waiting"
    });
}

// Sort: Female first, then Male
players.sort((a, b) => {
    if (a.gender === "Female" && b.gender !== "Female") return -1;
    if (a.gender !== "Female" && b.gender === "Female") return 1;
    return a.name.localeCompare(b.name);
});

console.log(JSON.stringify(players, null, 2));
